monix-execution/shared/src/main/scala_2.13+/monix/execution/internal/compat/package.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2014-2019 by The Monix Project Developers.
+ * See the project homepage at: https://monix.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package monix.execution.internal
+
+import scala.reflect.ClassTag
+
+package object compat {
```

Comment:
The problem is that we cannot expose types in sub-package `internal` that are not `private[monix]`.

Also regarding `Traversable` or `TraversableOnce`, I'd rather us switch to `Iterable`, because things that aren't `Iterable` are also side effecting and thus error prone and I'd rather not deal with those.

This is also forward thinking. I don't want the collection library's authors to change their minds again. There's no `IterableOnce` in Java for example, there's only `Iterator`, I think `IterableOnce` is a case of inheritance gone wrong, so I'd like for us to not take that bait again 🙂

So screw it, lets switch to `Iterable` wherever we used `Traversable` or `TraversableOnce`.

Reply:
Done. Changing to `Iterable` over `TraversableOnce` does mean it is no longer possible to pass an `Iterator`, is that okay?

Reply:
Yes, should be totally fine — I think that people working with `Iterator` can always build a factory (and thus an `Iterable` for it) plus you can always do a `.toIterable`, which I'm assuming works in Scala 2.13 too:

```scala
scala> val it = List(1,2,3,4,5).iterator
it: Iterator[Int] = <iterator>

scala> it.toIterable
res0: Iterable[Int] = Stream(1, ?)
```